csharp/src/Ice/LocalException.cs

```diff
@@ -138,7 +170,9 @@ public ConnectFailedException(Exception innerException)
     public class ConnectTimeoutException : ConnectFailedException
     {
         /// <summary>Constructs a new instance of the <see cref="ConnectTimeoutException"/> class.</summary>
-        public ConnectTimeoutException()
+        /// <param name="connection">The connection associated with this failure.</param>
+        public ConnectTimeoutException(Connection connection)
+            : base(RetryPolicy.AfterDelay(TimeSpan.Zero), connection)
```

Comment:
This baked-in RetryPolicy is inconsistent with what we do for remote exceptions.

For remote exceptions the retry policy is never baked in, it's whoever throws the exception that decides on the retry policy (default = no retry), and I think it should be the same for local exceptions.

If we wanted a baked-in retry policy, then it would be expressed through different means like a base "marker" interface.